scalding-core/src/main/scala/com/twitter/scalding/serialization/StringOrderedSerialization.scala

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2015 Twitter, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package com.twitter.scalding.serialization
+
+import java.io.{ InputStream, OutputStream }
+import scala.util.{ Failure, Success, Try }
+import scala.util.control.NonFatal
+
+import JavaStreamEnrichments._
+
+object StringOrderedSerialization {
+  final def binaryIntCompare(leftSize: Int, seekingLeft: InputStream, rightSize: Int, seekingRight: InputStream): Int = {
```

Comment:
This works because UTF-8 interpreted as a series of *unsigned* bytes sorts correctly when sorted on its bytes, right?

What's the advantage of comparing as many longs as we can, then ints, then bytes? Logically comparing just bytes would be correct, right? So there's some benefit to using the largest type that we can?